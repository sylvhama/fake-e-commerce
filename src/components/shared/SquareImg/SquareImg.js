import React from 'react';
import styled from 'styled-components';

const Square = styled.figure`
  position: relative;
  margin: 0;
  padding-top: 100%;

  & img {
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    object-fit: contain;
  }
`;

export default ({ src, alt, ...rest }) => (
  <Square {...rest}>
    <img src={src} alt={alt} />
  </Square>
);
