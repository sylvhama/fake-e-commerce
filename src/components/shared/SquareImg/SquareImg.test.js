import React from 'react';
import 'jest-styled-components';

import { renderWithTheme } from '../../../tests/helpers';

import SquareImg from './SquareImg';

test('SquareImg renders correctly', () => {
  const rendered = renderWithTheme(<SquareImg src="src" alt="alt" />).toJSON();
  expect(rendered).toMatchSnapshot();
});
