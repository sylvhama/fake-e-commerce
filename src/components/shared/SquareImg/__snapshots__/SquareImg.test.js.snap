// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquareImg renders correctly 1`] = `
.c0 {
  position: relative;
  margin: 0;
  padding-top: 100%;
}

.c0 img {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  object-fit: contain;
}

<figure
  className="c0"
>
  <img
    alt="alt"
    src="src"
  />
</figure>
`;
